### PR TITLE
fix #69

### LINKED ISSUE #69: Failed to get EnhancedClassInstanceContext when the class has no constructor point

### PR DESCRIPTION
**Pull request by ascrutae**

### CODE REVIEW

File: skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java [hunk 1]
```diff
@@ -70,8 +69,15 @@
...
-                        MethodDelegation.to(new ClassConstructorInterceptor(constructorInterceptPoint.getConstructorInterceptor()))
-                                .appendParameterBinder(FieldProxy.Binder.install(FieldGetter.class, FieldSetter.class))));
+                                MethodDelegation.to(new ClassConstructorInterceptor(constructorInterceptPoint.getConstructorInterceptor()))
+                                        .appendParameterBinder(FieldProxy.Binder.install(FieldGetter.class, FieldSetter.class))));
+            }
+        } else {
+            for (ConstructorInterceptPoint constructorInterceptPoint : constructorInterceptPoints) {
+                newClassBuilder = newClassBuilder.constructor(ElementMatchers.<MethodDescription>any())
+                        .intercept(SuperMethodCall.INSTANCE.andThen(
+                                MethodDelegation.to(new DefaultClassConstructorInterceptor(constructorInterceptPoint.getConstructorInterceptor()))
```
Comment (raphw) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: If you want to be sure, always define the `any` matching for constructors first and then apply any potential custom interception. Any element machter that is registered later will override the previous ones.

Comment (wu-sheng) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: @raphw , is there only one interceptor for each constructor? It's only override, not as a chain?

File: skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java [hunk 2]
```diff
@@ -70,8 +69,15 @@
             for (ConstructorInterceptPoint constructorInterceptPoint : constructorInterceptPoints) {
                 newClassBuilder = newClassBuilder.constructor(constructorInterceptPoint.getConstructorMatcher())
```
Comment (wu-sheng) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: @raphw, is there other way to init a attribute, which is type of EnhancedClassInstanceContext? Maybe not use a custom interception?

Comment (wu-sheng) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: @raphw , I have notice your response on https://github.com/raphw/byte-buddy/issues/227. Maybe there is no other way?

File: skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java [hunk 3]
```diff
@@ -70,8 +69,15 @@
             for (ConstructorInterceptPoint constructorInterceptPoint : constructorInterceptPoints) {
```
Comment (wu-sheng) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: @ascrutae , before iterator, set the DefaultInterceptor on `any()` matcher, to ensure, EnhancedClassInstanceContext attr initialize always。

File: skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java (hunk 1 above)
Comment (raphw) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: No, you would need to chain them for this, the last one intercepts the next one. One way to concatenate interceptions would be to use Byte Buddy's `Advice` as a visitor where they are grouped.

Comment (wu-sheng) [inline skywalking-sniffer/skywalking-api/src/main/java/com/a/eye/skywalking/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: Right now, I maybe not need the chain feature. But I would like to learn how to use it. Can you demonstrate for me?

**Pull request merged**